src/AppInstallerCLICore/Workflows/PortableInstallFlow.cpp

```diff
@@ -0,0 +1,379 @@
+                    }
+                    else
+                    {
+                        AICLI_LOG(CLI, Info, << "Overriding registry match check...");
+                        context.Reporter.Warn() << Resource::String::PortableRegistryCollisionOverridden << std::endl;
+                    }
+                }
+            }
+
+            AICLI_LOG(CLI, Info, << "Begin writing to Uninstall registry.");
+            uninstallEntry.SetValue(PortableValueName::DisplayName, entry.DisplayName);
+            uninstallEntry.SetValue(PortableValueName::DisplayVersion, entry.DisplayVersion);
+            uninstallEntry.SetValue(PortableValueName::Publisher, entry.Publisher);
+            uninstallEntry.SetValue(PortableValueName::InstallDate, Utility::GetCurrentDateForARP());
+            uninstallEntry.SetValue(PortableValueName::URLInfoAbout, manifest.DefaultLocalization.Get<Manifest::Localization::PackageUrl>());
+            uninstallEntry.SetValue(PortableValueName::HelpLink, manifest.DefaultLocalization.Get<Manifest::Localization::PublisherSupportUrl>());
+            uninstallEntry.SetValue(PortableValueName::UninstallString, L"winget uninstall --product-code " + productCode);
+            uninstallEntry.SetValue(PortableValueName::WinGetInstallerType, ConvertToUTF16(InstallerTypeToString(InstallerTypeEnum::Portable)));
+            uninstallEntry.SetValue(PortableValueName::WinGetPackageIdentifier, manifest.Id);
+            uninstallEntry.SetValue(PortableValueName::WinGetSourceIdentifier, sourceIdentifier);
+            uninstallEntry.SetValue(PortableValueName::PortableTargetFullPath, GetPortableTargetFullPath(context).wstring());
+            uninstallEntry.SetValue(PortableValueName::PortableSymlinkFullPath, GetPortableSymlinkFullPath(context).wstring());
+            uninstallEntry.SetValue(PortableValueName::SHA256, Utility::SHA256::ConvertToWideString(context.Get<Execution::Data::HashPair>().second));
+            uninstallEntry.SetValue(PortableValueName::InstallLocation, GetPortableTargetDirectory(context).wstring());
+            AICLI_LOG(CLI, Info, << "Writing to Uninstall registry complete.");
+        }
+
+        void MovePortableExeAndCreateSymlink(Execution::Context& context)
+        {
+            const std::filesystem::path& installerPath = context.Get<Execution::Data::InstallerPath>();
+            const std::filesystem::path& targetFullPath = GetPortableTargetFullPath(context);
+            const std::filesystem::path& symlinkFullPath = GetPortableSymlinkFullPath(context);
+            const std::filesystem::path& targetDirectory = GetPortableTargetDirectory(context);
+
+            bool isDirectoryCreated = false;
+            if (!std::filesystem::exists(targetDirectory) || !std::filesystem::is_directory(targetDirectory))
+            {
+                if (!std::filesystem::create_directories(targetDirectory))

```

Comment:
Do you disagree with me that one should be able to install a portable to an existing directory?  Because my reading of the return value of `std::filesystem::create_directories` is that it will be false if `targetDirectory` already exists.  It will just throw on an error.

If one does not exist, please add a test for installing to an existing directory.